### PR TITLE
MGCFD-OP2: New Package

### PR DESCRIPTION
**Pull request by jessj0nes**
OP2 port of MGCFD: A 3D unstructured multigrid, finite-volume computational fluid dynamics (CFD) mini-app for inviscid-flow.

Package testing conducted on AWS Graviton 2 and Graviton 3 with GCC 12.1.0, Arm 22.0.1 and NVHPC 22.3 compilers.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mgcfd-op2/package.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            "gcc": "gnu",
+            "arm": "clang",
+            "cce": "cray",
+            "nvhpc": "pgi",
+        }
+
+        if self.spec.compiler.name in compiler_map:
+            env["COMPILER"] = compiler_map[self.spec.compiler.name]
+        else:
+            env["COMPILER"] = self.spec.compiler.name
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mgcfd-op2/package.py]: This is better suited for `setup_build_environment`

File: var/spack/repos/builtin/packages/mgcfd-op2/package.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        else:
+            builds = ["seq", "openmp"]
+            if "+cuda" in self.spec and spec.variants["cuda_arch"].value[0] != "none":
+                builds.append("cuda")
+        return builds
+
+    def build(self, spec, prefix):
+        for b in self.get_builds():
+            make("clean_" + b)
+            make(b)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mgcfd-op2/package.py]: Why is it nbecessary to `clean_` from a fresh stage? Also, I think you can just define the `build_targets` property.

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by alalazo** [comment]
@bozbez @tomdeakin @gihanmudalige Can you confirm it's fine with you to be listed as maintainers of this package?

**Comment by bozbez** [comment]
Yeah that's fine.

**Comment by tomdeakin** [comment]
> @bozbez @tomdeakin @gihanmudalige Can you confirm it's fine with you to be listed as maintainers of this package?

Happy to be listed as a maintainer.

**Comment by gihanmudalige** [comment]
Yes. I am happy to be listed here as well.

File: var/spack/repos/builtin/packages/mgcfd-op2/package.py (hunk 1 above)
Comment (jessj0nes) [reply var/spack/repos/builtin/packages/mgcfd-op2/package.py]: Done - thanks

File: var/spack/repos/builtin/packages/mgcfd-op2/package.py (hunk 2 above)
Comment (jessj0nes) [reply var/spack/repos/builtin/packages/mgcfd-op2/package.py]: Thanks for the feedback, I'm new to writing these packages so appreciate the refinement. Your suggestions are a great improvement!

File: var/spack/repos/builtin/packages/mgcfd-op2/package.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    homepage = "https://github.com/warwick-hpsc/MG-CFD-app-OP2"
+    git = "https://github.com/warwick-hpsc/MG-CFD-app-OP2.git"
+
+    maintainers = ["tomdeakin", "gihanmudalige", "bozbez"]
+
+    version("v1.0.0-rc1")
+
+    variant("mpi", default=False, description="Enable MPI support")
+
+    depends_on("gmake@4.3")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mgcfd-op2/package.py]:
```suggestion
    depends_on("gmake@4.3:")
```

**Review by jessj0nes — COMMENTED** [review]
Should be good to go

**Review by tomdeakin — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**